test/integration/src/test/java/io/pravega/test/integration/KVTableWithStreamTest.java

```diff
@@ -0,0 +1,769 @@
+                Futures.getAndHandleExceptions(this.controller.getCurrentSegments(newScope1, newStream1),
+                        RuntimeException::new).getSegments().size());
+        assertEquals("The stream segment count should be same as passed", streamSegments,
+                Futures.getAndHandleExceptions(this.controller.getCurrentSegments(newScope1, newStream2),
+                        RuntimeException::new).getSegments().size());
+
+        Iterator<KeyValueTableInfo> tableItr1 = this.controller.listKeyValueTables(newScope1).asIterator();
+        assertFalse("Table List Iterator 1 with streams should not have any table entries in it",
+                tableItr1.hasNext());
+        AssertExtensions.assertThrows(NoSuchElementException.class, () -> tableItr1.next());
+
+        // test Scope2.
+        assertTrue("The requested Scope must get created", this.controller.createScope(newScope2).join());
+        Iterator<Stream> streamItr2 = this.controller.listStreams(newScope2).asIterator();
+        assertFalse("Stream list iterator2 should be empty", streamItr2.hasNext());
+
+        Iterator<KeyValueTableInfo> tableItr2 = this.controller.listKeyValueTables(newScope2).asIterator();
+        assertFalse("Table List Iterator 2 without streams should not have any table entries in it",
+                tableItr2.hasNext());
+        AssertExtensions.assertThrows(NoSuchElementException.class, () -> tableItr2.next());
+
+        // seal stream1.
+        CompletableFuture<Boolean> sealStreamStatus = controller.sealStream(newScope1, newStream1);
+        log.info("Sealing stream1 {}", newStream1);
+        assertTrue(sealStreamStatus.get());
+        // seal stream2.
+        sealStreamStatus = controller.sealStream(newScope1, newStream2);
+        log.info("Sealing stream2 {}", newStream2);
+        assertTrue(sealStreamStatus.get());
+        // delete stream1.
+        CompletableFuture<Boolean> deleteStreamStatus = controller.deleteStream(newScope1, newStream1);
+        log.info("Deleting stream1 {}", newStream1);
+        assertTrue(deleteStreamStatus.get());
+        // delete stream2.
+        deleteStreamStatus = controller.deleteStream(newScope1, newStream2);
+        log.info("Deleting stream2 {}", newStream2);
+        assertTrue(deleteStreamStatus.get());
+        // delete scope1.
+        CompletableFuture<Boolean> deleteScopeStatus = controller.deleteScope(newScope1);
+        log.info("Deleting scope1 {}", newScope1);
+        assertTrue(deleteScopeStatus.get());
+        // delete scope2.
+        deleteScopeStatus = controller.deleteScope(newScope2);
+        log.info("Deleting scope2 {}", newScope2);
+        assertTrue(deleteScopeStatus.get());
+
+        log.info("Successfully completed testing list KVTables under a scope with " +
+                "only Streams and under empty Scope ");
+    }
+
+    private Integer getKeyID() {
+        Integer keyId = random.nextInt(endKeyRange - startKeyRange) + startKeyRange;
```

Comment:
You do not need to declare a variable here.

Reply:
I'll refactor the code here. Thanks